go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -139,16 +186,24 @@ func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add bo
 								cookie, npw.ofportPhys, flowProtocol, svcPort.NodePort, HostNodePortCTZone, npw.gatewayIPv4, svcPort.TargetPort.String()))
 					}
 					nodeportFlows = append(nodeportFlows,
-						// table 6, Sends the packet to the host
+						// table 6, Sends the packet to the host. Note that the constant etp svc cookie is used since this flow would be
+						// same for all such services.
 						fmt.Sprintf("cookie=%s, priority=110, table=6, actions=output:LOCAL",
-							cookie),
+							etpSvcOpenFlowCookie),
 						// table 0, Matches on return traffic, i.e traffic coming from the host networked pod's port, and unDNATs
 						fmt.Sprintf("cookie=%s, priority=110, in_port=LOCAL, %s, tp_src=%s, actions=ct(zone=%d nat,table=7)",
 							cookie, flowProtocol, svcPort.TargetPort.String(), HostNodePortCTZone),
-						// table 7, Sends the packet back out eth0 to the external client
+						// table 7, Sends the packet back out eth0 to the external client. Note that the constant etp svc
+						// cookie is used since this would be same for all such services.
 						fmt.Sprintf("cookie=%s, priority=110, table=7, "+
-							"actions=output:%s", cookie, npw.ofportPhys))
+							"actions=output:%s", etpSvcOpenFlowCookie, npw.ofportPhys))
 
+					npw.ofm.updateFlowCacheEntry(key, nodeportFlows)
+				} else if isServiceTypeETPLocal && epNonHostLocal && config.Gateway.Mode == config.GatewayModeLocal {
```

Comment:
i'm not sure this is correct. From what I understand this is saying: If service is ETP Local && there are ovn networked local endpoints && local gateway, then send it LOCAL. If that is not true, then it will go to the else which will add shared gw type flows to send the packet to OVN right? Is that what we want? We would want OVN to send the TCP reset if it was a TCP packet. I guess it should be fine, but should also be fine to send it into the host whether epNonHostLocal is true or not, and then the reject would happen once the local stack sends it into mp0? Might be better for consistency.

Reply:
> i'm not sure this is correct. From what I understand this is saying: If service is ETP Local && there are ovn networked local endpoints && local gateway, then send it LOCAL. If that is not true, then it will go to the else which will add shared gw type flows to send the packet to OVN right? Is that what we want? 

yes that's what we want, if its shared gateway then we'd need it to go ovn and then it hits the `skip_snat=true` LB etc.


> We would want OVN to send the TCP reset if it was a TCP packet. I guess it should be fine, but should also be fine to send it into the host whether epNonHostLocal is true or not, and then the reject would happen once the local stack sends it into mp0? Might be better for consistency.

 I am not sure I follow....